### DISCUSSION TITLE
Managing the partition in Db?

### DISCUSSION THREAD

**Discussion by apoorv-98** [milvus-io/milvus#29748] [Q&A and General discussion] [answered]
In milvus when you load a collection you need to pass a partition name, i want to load all the partition present in a collection of the db how can i achieve this and i want to do query or searching to the specific partition of Db?

**Comment by yhmo** [comment]
To load a collection entirely into memory:
`collection.load()`

To load a specific partition into memory:
`collection.load(partition_names=["XXX"])`

The definition of load() interface in the python sdk, the parameter "partition_names" is optional.
```
    def load(
        self,
        partition_names: Optional[list] = None,
        replica_number: int = 1,
        timeout: Optional[float] = None,
        **kwargs,
    ):
```

**Reply by yhmo** [reply, marked as answer]
To search/query in a specific partition:
```
collection.query(expr="xxxxxx", partition_names=["XXX"])
collection.search(data=vectors, param={}, partition_names=["XXX"])
```

Both of the query() and search() have the optional parameter "partition_names"

**Reply by apoorv-98** [reply]
thanks @yhmo for your response.
One more follow up query.
If we load the collection without mentioning the partition name it will load all the partition of that collection?

**Reply by yhmo** [reply]
Right, collection.load() will load all the partitions of the collection.
